.github/workflows/server-test.yml

```diff
@@ -0,0 +1,49 @@
+# Server test scenario
+name: Server Integration Tests
+
+# FIXME put only necessary triggers
```

Comment:
@ggerganov Which triggers do you suggest ?

Reply:
Maybe like so:

```yml
on:
  workflow_dispatch: # allows manual triggering
  push:
    branches:
      - master
    paths: ['.github/workflows/**', '**/CMakeLists.txt', '**/Makefile', '**/*.h', '**/*.hpp', '**/*.c', '**/*.cpp']
  pull_request:
    types: [opened, synchronize, reopened]
    paths: ['**/CMakeLists.txt', '**/Makefile', '**/*.h', '**/*.hpp', '**/*.c', '**/*.cpp']
```